.scripts/checkIfPublished.mjs

```diff
@@ -0,0 +1,22 @@
+import { main } from "effection";
+import { fetch } from "@effection/fetch";
+
+const pkgName = process.argv[2];
+const tag = process.argv[3];
+
+main(function* () {
+  const pkg = yield packageExists(pkgName, tag);
+  console.log(pkg);
+});
+
+function* packageExists(name, tag) {
+  let request = yield fetch(`https://registry.npmjs.com/${name}`);
+  let response = yield request.json();
+  if (request.status === 404) {
+    return "not published";
+  } else if (request.status < 400) {
+    return response["dist-tags"][tag];
+  } else {
+    throw new Error("request error");
+  }
```

Comment:
The `fetch` operation completes once all headers have been received, so no need to pull the body and parse as JSON unless we're going to need it. Also, because the body might be invalid json on a `404` this might be more correct.

```suggestion
  let request = yield fetch(`https://registry.npmjs.com/${name}`);
  if (request.status === 404) {
    return "not published";
  } else if (request.status < 400) {
    let response = yield request.json();
    return response["dist-tags"][tag];
  } else {
    throw new Error("request error");
  }
```

Reply:
ooh, yea that makes a lot of sense. Thanks!